src/sentry/eventstore/models.py

```diff
@@ -716,18 +716,18 @@ def __init__(
         data: NodeData,
         snuba_data: Mapping[str, Any] | None = None,
         occurrence: IssueOccurrence | None = None,
-    ):
+    ) -> None:
         super().__init__(project_id, event_id, snuba_data=snuba_data)
         self.group = group
         self.data = data
         self._occurrence = occurrence
 
-    def __eq__(self, other):
+    def __eq__(self, other: BaseEvent) -> bool:
```

Comment:
`__eq__` should always have the signature `def __eq__(self, other: object) -> bool:`